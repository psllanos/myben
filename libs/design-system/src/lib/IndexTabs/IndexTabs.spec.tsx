import { render, screen } from '@testing-library/react'
import { useRef } from 'react'
import { IndexTabs } from './'

const Example = () => {
    const scrollContainer = useRef<HTMLDivElement>(null)
    const section1 = useRef<HTMLDivElement>(null)
    const section2 = useRef<HTMLDivElement>(null)

    return (
        <>
            <IndexTabs
                scrollContainer={scrollContainer}
                sections={[
                    {
                        name: 'Section 1',
                        elementRef: section1,
                    },
                    {
                        name: 'Section 2',
                        elementRef: section2,
                    },
                ]}
            />
            <div style={{ overflowY: 'auto', height: '400px' }} ref={scrollContainer}>
                <div ref={section1} style={{ height: '500px' }}>
                    Content 1
                </div>
                <div ref={section2} style={{ height: '500px' }}>
                    Content 2
                </div>
            </div>
        </>
    )
}

// Very light testing - it's challenging to mock/test all of the scroll interactions
describe('IndexTabs', () => {
    it('should render correctly', () => {
        const component = render(<Example />)

        expect(screen.getByText('Section 1')).toBeInTheDocument()
        expect(screen.queryByText('Content 1')).toBeInTheDocument()

        expect(screen.getByText('Section 2')).toBeInTheDocument()
        expect(screen.queryByText('Content 2')).toBeInTheDocument()

        expect(component).toMatchSnapshot()
    })
})
