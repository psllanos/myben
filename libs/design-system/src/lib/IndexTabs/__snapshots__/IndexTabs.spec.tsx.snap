// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexTabs should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="relative flex gap-2 w-full overflow-x-scroll scrollbar-none text-base"
      >
        <button
          class="py-1 px-2 rounded-md whitespace-nowrap hover:bg-gray-600 transition text-white bg-gray-500"
        >
          Section 1
        </button>
        <button
          class="py-1 px-2 rounded-md whitespace-nowrap hover:bg-gray-600 transition text-gray-100"
        >
          Section 2
        </button>
      </nav>
      <div
        style="overflow-y: auto; height: 400px;"
      >
        <div
          style="height: 500px;"
        >
          Content 1
        </div>
        <div
          style="height: 500px;"
        >
          Content 2
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="relative flex gap-2 w-full overflow-x-scroll scrollbar-none text-base"
    >
      <button
        class="py-1 px-2 rounded-md whitespace-nowrap hover:bg-gray-600 transition text-white bg-gray-500"
      >
        Section 1
      </button>
      <button
        class="py-1 px-2 rounded-md whitespace-nowrap hover:bg-gray-600 transition text-gray-100"
      >
        Section 2
      </button>
    </nav>
    <div
      style="overflow-y: auto; height: 400px;"
    >
      <div
        style="height: 500px;"
      >
        Content 1
      </div>
      <div
        style="height: 500px;"
      >
        Content 2
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
